apps/web/app/api/(internal)/pipeline/lib/telemetry.ts

```diff
@@ -0,0 +1,178 @@
+import { IntegrationType } from "@prisma/client";
+import { createHash } from "node:crypto";
+import { type CacheKey, getCacheService } from "@formbricks/cache";
+import { prisma } from "@formbricks/database";
+import { logger } from "@formbricks/logger";
+import { env } from "@/lib/env";
+import packageJson from "../../../../../package.json";
+
+const TELEMETRY_INTERVAL_MS = 24 * 60 * 60 * 1000; // 24 hours
+const TELEMETRY_LOCK_KEY = "telemetry_lock" as CacheKey;
+const TELEMETRY_LAST_SENT_KEY = "telemetry_last_sent_ts" as CacheKey;
+
+let nextTelemetryCheck = 0;
+
+export const sendTelemetryEvents = async () => {
+  try {
+    const now = Date.now();
+    // 1. In-memory check
+    if (now < nextTelemetryCheck) {
+      return;
+    }
+
+    // 2. Redis check
+    const cacheServiceResult = await getCacheService();
+    if (!cacheServiceResult.ok) {
+      // Fallback: update next check to try again in 1 hour if cache fails, to avoid spamming
+      nextTelemetryCheck = now + 60 * 60 * 1000;
+      return;
+    }
+    const cache = cacheServiceResult.data;
+
+    const lastSentResult = await cache.get(TELEMETRY_LAST_SENT_KEY);
+    const lastSentStr = lastSentResult.ok && lastSentResult.data ? (lastSentResult.data as string) : null;
+    const lastSent = lastSentStr ? Number.parseInt(lastSentStr, 10) : 0;
+
+    if (now - lastSent < TELEMETRY_INTERVAL_MS) {
+      // Update in-memory check to match the remaining time
+      nextTelemetryCheck = lastSent + TELEMETRY_INTERVAL_MS;
+      return;
+    }
+
+    // 3. Acquire Lock to prevent concurrent execution in cluster
+    const redis = cache.getRedisClient();
+    if (!redis) {
+      return;
+    }
+
+    const acquired = await redis.set(TELEMETRY_LOCK_KEY, "locked", {
+      NX: true,
+      PX: 5 * 60 * 1000, // 5 minutes
```

Comment:
5 minutes is quite long for a short operation like this. I think one minute should be enough here. What do you think?
The only risk of a long lock time is, that the instance which holds the lock crashes and blocks every other instance for longer time.